OpenRA.Mods.Common/Traits/BotModules/Squads/States/StateBase.cs

```diff
@@ -106,19 +106,12 @@ protected virtual bool ShouldFlee(Squad squad, Func<IReadOnlyCollection<Actor>,
 
 		protected static bool IsRearming(Actor a)
 		{
-			if (a.IsIdle)
-				return false;
-
 			var activity = a.CurrentActivity;
-			if (activity.GetType() == typeof(Resupply))
-				return true;
-
-			var next = activity.NextActivity;
-			if (next == null)
-				return false;
-
-			if (next.GetType() == typeof(Resupply))
-				return true;
+			for (; activity != null; activity = activity.ChildActivity)
```

Comment:
Use
```csharp
if (a.CurrentActivity.ActivitiesImplementing<Resupply>().Any())
    return true;
```

and revert the change to Activity.cs (`ChildActivity` is deliberately kept non-public to avoid mod code from making a mess with its internals).

Reply:
looks like we also need a null check here.

Reply:
fixed